internal/services/containers/container_registry_task_resource.go

```diff
@@ -0,0 +1,1573 @@
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ContainerRegistryTaskName,
+		},
+		"container_registry_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.RegistryID,
+		},
+		"platform_setting": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*schema.Schema{
+					"os": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(legacyacr.Windows),
+							string(legacyacr.Linux),
+						}, false),
+					},
+					"architecture": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(legacyacr.Amd64),
+							string(legacyacr.Arm),
+							string(legacyacr.Arm64),
+							string(legacyacr.ThreeEightSix),
+							string(legacyacr.X86),
+						}, false),
+					},
+					"variant": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(legacyacr.V6),
+							string(legacyacr.V7),
+							string(legacyacr.V8),
+						}, false),
+					},
+				},
+			},
+		},
+		"docker_step": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*schema.Schema{
+					"dockerfile_path": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+					"context_path": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+					"context_access_token": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						Sensitive:    true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+					"image_names": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						Elem: &pluginsdk.Schema{
+							Type: pluginsdk.TypeString,
+						},
+					},
+					"is_push_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+						Default:  true,
+					},
+					"is_cache_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+						Default:  true,
+					},
+					"target": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+					"arguments": {
+						Type:     pluginsdk.TypeMap,
+						Optional: true,
+						Elem: &pluginsdk.Schema{
+							Type: pluginsdk.TypeString,
+						},
+					},
+					"secret_arguments": {
+						Type:      pluginsdk.TypeMap,
+						Optional:  true,
+						Sensitive: true,
+						Elem: &pluginsdk.Schema{
+							Type: pluginsdk.TypeString,
+						},
+					},
+				},
+			},
+			ConflictsWith: []string{"file_task_step", "encoded_task_step"},
+		},
+		"file_task_step": {
```

Comment:
```suggestion
		"file": {
```

task is in the resource name

Reply:
going off your comment above: 
```suggestion
		"file_step": {
```